pilot/pkg/model/service.go

```diff
@@ -591,6 +602,29 @@ func (ep *IstioEndpoint) CmpOpts() []cmp.Option {
 	return istioEndpointCmpOpts
 }
 
+// SortIstioEndpointByAddresses returns the IstioEndpoint address slices sorted by its addresses.
+func (ep *IstioEndpoint) SortIstioEndpointByAddresses() []string {
+	if ep == nil {
+		return nil
+	}
+
+	addrSlice := ep.Addresses
+	sort.Sort(sort.Reverse(sort.StringSlice(addrSlice)))
+	return addrSlice
+}
+
+// GetIstioEndpointKey returns the key of IstioEndpoint based on its addresses
+func (ep *IstioEndpoint) GetIstioEndpointKey() string {
+	if ep == nil {
+		return ""
+	}
+
+	addrSlice := ep.Addresses
+	sort.Sort(sort.Reverse(sort.StringSlice(addrSlice)))
```

Comment:
Why reverse? Why do we need to consider all addresses instead of just the primary one?

Reply:
Hi @howardjohn, this should be opened, assume that
endpoint 1 :  [127.0.0.1, ::1] and endpoint 2:  [::1, 127.0.01]
shall endpoint 1 and endpoint 2 be considered as the same endpoint or not?
IMO, it should be the same, so I sort them to keep they are the same one. any idea?